framework/SupervisedLearning/ARMA.py

```diff
@@ -259,6 +262,12 @@ def __trainLocal__(self,featureVals,targetVals):
     for t,target in enumerate(self.target):
       timeSeriesData = targetVals[:,t]
       self._signalStorage[target]['original'] = copy.deepcopy(timeSeriesData)
+      # if we're enforcing the training CDF, we should store it now
+      if self.preserveInputCDF:
+        counts, edges = np.histogram(timeSeriesData, bins=self._computeNumberOfBins(timeSeriesData), density=False)
+        counts = np.asarray(counts) / float(len(timeSeriesData))
+        dist = stats.rv_histogram((counts, edges))
+        self._trainingCDF[target] = dist
```

Comment:
several comments:
1. I think you may need to create a utils functions to compute the cdf distributions
2. Joshua created the MetricUtilities.py to convert weighted data into a PDF and CDF function, I think it is worth to checking it to avoid some duplicated functions.

Reply:
Josh's implementation in MetricUtilities is local to that folder, and doesn't contain the "ppf" (inverse CDF) as far as I can tell reading through it. I can extract this to a utility in mathUtils and make a request to expand and generalize the MetricUtilities method.

Reply:
I leave the design to the individual who will address the issue. If it is me, then I think making a Distribution class that holds all the typical Distribution methods (including pdf, cdf, rvs, ppf, etc) makes the most sense. That way the weighted distribution information Josh created can be preserved and all the utility needed elsewhere can be met.